### PR TITLE
Fix bugs from PE-D

### LINKED ISSUE #115: [PE-D][Tester B] Able to insert negative weights when adding a logging a new exercise
Should not be allowed to log negative weights

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: JustinSoh/ped#4

### PR DESCRIPTION
**Pull request by pqienso**
fixes #115, #116, #129, #109

### CODE REVIEW

File: src/main/java/byteceps/processing/ExerciseManager.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ public String execute(Parser parser) throws Exceptions.InvalidInput,
     private String executeEditAction(Parser parser) throws Exceptions.InvalidInput, Exceptions.ActivityDoesNotExists {
         String newExerciseName = processEditExercise(parser,this);
         return String.format(
-                ManagerStrings.EXERCISE_EDITED, parser.getActionParameter(), newExerciseName
+                ManagerStrings.EXERCISE_EDITED, parser.getActionParameter().toLowerCase(), newExerciseName
```
Comment (LWachtel1) [inline src/main/java/byteceps/processing/ExerciseManager.java:66]: Good fix for [PE-D][Tester D] Case-Sensitive exercise #116

File: src/main/java/byteceps/processing/ExerciseManager.java [hunk 2]
```diff
@@ -94,7 +94,7 @@ private String executeAddAction(Parser parser) throws Exceptions.InvalidInput,
 
     //@@author V4vern
     private Exercise processAddExercise(Parser parser) throws Exceptions.InvalidInput {
-        String exerciseName = parser.getActionParameter();
+        String exerciseName = parser.getActionParameter().toLowerCase();
```
Comment (LWachtel1) [inline src/main/java/byteceps/processing/ExerciseManager.java:97]: Good fix for [PE-D][Tester D] Case-Sensitive exercise #116

File: src/main/java/byteceps/processing/WorkoutLogsManager.java [hunk 3]
```diff
@@ -37,6 +37,10 @@ public void addExerciseLog(String workoutLogDate, String exerciseName,
             int setsInt = Integer.parseInt(sets);
             int repsInt = Integer.parseInt(repetitions);
 
+            if (weightInt < 0 || setsInt < 0 || repsInt < 0) {
+                throw new NumberFormatException();
+            }
+
```
Comment (LWachtel1) [inline src/main/java/byteceps/processing/WorkoutLogsManager.java:43]: Good fix for [PE-D][Tester B] Able to insert negative weights when adding a logging a new exercise #115

File: src/main/java/byteceps/storage/Storage.java [hunk 4]
```diff
@@ -143,6 +152,14 @@ private void loadWorkoutLogs(ExerciseManager allExercises, WorkoutManager allWor
...
             String workoutDate = currentWorkout.getString(StorageStrings.WORKOUT_DATE);
             String workoutName = currentWorkout.getString(StorageStrings.WORKOUT_NAME);
+            //validate that workoutDate is a valid date string
+            try {
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DayStrings.YEAR_FORMAT);
+                formatter = formatter.withLocale(formatter.getLocale());
+                LocalDate.parse(workoutDate, formatter); //ignore result, just catch exception
+            } catch (DateTimeParseException e) {
+                throw new Exceptions.InvalidInput(""); //no need for error message, LOAD_ERROR will be printed
+            }
```
Comment (LWachtel1) [inline src/main/java/byteceps/storage/Storage.java:162]: Good fix for [PE-D][Tester B] Changing the data.json to have an invalid date crashes the program #129

**Review by LWachtel1 — COMMENTED** [review]
LGTM - waiting on second opinion just due to number of issues this PR resolves

**Comment by V4Vern** [comment]
LGTM

File: src/test/java/byteceps/storage/StorageTest.java [hunk 5]
```diff
@@ -204,7 +205,7 @@ public void execute_load_corruptedJSONFailure() {
 
         assertDoesNotThrow( () -> checkFile(corruptFile));
 
-        Storage failureStorage = new Storage(corruptFile);
+        Storage failureStorage = new Storage(corruptFile, new UserInterface());
```
Comment (joshualeejunyi) [inline src/test/java/byteceps/storage/StorageTest.java:208]: Are we able to use the storage instance that we have in the setup() method?

Comment (LWachtel1) [reply src/test/java/byteceps/storage/StorageTest.java:208]: The storage instance in setup() uses data.json by default. This failure test case needs to corrupted.json for it to function. Apologies

**Pull request merged**